src/routers/alpha-router/alpha-router.ts

```diff
@@ -1995,17 +2019,33 @@ export class AlphaRouter
         )
       : Promise.resolve(null);
 
-    const [usdPool, nativeQuoteTokenV3Pool, nativeAmountTokenV3Pool] =
-      await Promise.all([
-        usdPoolPromise,
-        nativeQuoteTokenV3PoolPromise,
-        nativeAmountTokenV3PoolPromise,
-      ]);
+    const nativeGasTokenV3PoolPromise =
+      providerConfig?.gasToken &&
```

Comment:
nit: I think it can be shorten to `!nativeCurrency .equals(providerConfig?.gasToken)`

Reply:
`.equals` doesn't seem to allow for undefined unfortunately so I think I still have to check for definedness